### PR TITLE
fix: Add a min cmake version for the cpp app

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fix: Add a min cmake version for the cpp app so that the user is prompted to install the latest cmake version instead of getting an ambiguous error

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.20)
```
Comment (osmaczko) [inline CMakeLists.txt:1]: I've version 3.18.4 and it works, so I think the minimum version should be 3.18 at most (not sure what is the real minimum version though, it didn't work with 3.16.x, right?)

**Review by osmaczko — CHANGES_REQUESTED** [review]
Thanks for that finding and fix :+1:

File: CMakeLists.txt (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply CMakeLists.txt:1]: Hey, yes 3.16 it didn't work.

if 3.18 works, shall do that :)

Comment (Khushboo-dev-cpp) [reply CMakeLists.txt]: done :)

**Review by osmaczko — APPROVED** [review]

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**